components/raftstore/src/store/region_snapshot.rs

```diff
@@ -23,26 +23,28 @@ use tikv_util::{panic_when_unexpected_key_or_data, set_panic_mark};
 ///
 /// Only data within a region can be accessed.
 #[derive(Debug)]
-pub struct RegionSnapshot<S: Snapshot> {
-    snap: Arc<S>,
+pub struct RegionSnapshot<EK>
+where
+    EK: KvEngine,
+{
+    snap: Arc<EK::Snapshot>,
     region: Arc<Region>,
     apply_index: Arc<AtomicU64>,
 }
 
-impl<S> RegionSnapshot<S>
+impl<EK> RegionSnapshot<EK>
 where
-    S: Snapshot,
+    EK: KvEngine,
 {
-    #[allow(clippy::new_ret_no_self)] // temporary until this returns RegionSnapshot<E>
-    pub fn new(ps: &PeerStorage<RocksEngine, RocksEngine>) -> RegionSnapshot<RocksSnapshot> {
+    pub fn new(ps: &PeerStorage<EK, RocksEngine>) -> RegionSnapshot<EK> {
```

Comment:
Could there use a type parameter 'ER' instead of 'RocksEngine'?

Reply:
Yes. I just haven't gotten to it yet.